Mathlib/RingTheory/LocalProperties/Semilocal.lean

```diff
@@ -0,0 +1,134 @@
+/-
+Copyright (c) 2025 Yiming Fu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yiming Fu
+-/
+import Mathlib.RingTheory.DedekindDomain.PID
+import Mathlib.RingTheory.KrullDimension.PID
+import Mathlib.RingTheory.Localization.Finiteness
+
+/-!
+# Local properties for semilocal rings
+
+This file proves some local properties for a semilocal ring `R` (a ring with
+finitely many maximal ideals).
+
+## Main results
+
+* `Module.finite.of_isLocalized_maximal`: A module `M` over a semilocal ring `R` is finite if it is
+  locally finite at every maximal ideal.
+* `isPrincipalIdealRing_of_isPrincipalIdealRing_localization`: A semilocal integral domain `A` is a
+  PID if its localization at every maximal ideal is a PID.
+-/
+
+variable {R : Type*} [CommSemiring R] [Finite (MaximalSpectrum R)]
+variable (M : Type*) [AddCommMonoid M] [Module R M]
+
+lemma Ring.krullDimLE_one_of_localization_maximal {A : Type*} [CommRing A] {n : ℕ}
+    (h : ∀ (P : Ideal A) [P.IsMaximal], Ring.KrullDimLE n (Localization P.primeCompl)) :
+    Ring.KrullDimLE n A := by
```

Comment:
Please move to [Mathlib.RingTheory.Ideal.Height](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Ideal/Height.html).